pennylane/templates/state_preparations/state_prep_mps.py

```diff
@@ -15,24 +15,100 @@
+        return mps
+
+    max_bond_dim = qml.math.max(d_shapes)
+
+    L = len(mps)
+    output_mps = [None] * L
+
+    # Procedure analogous to the left-canonical conversion but starting from the right and storing the Vd
+    for i in range(L - 1, 0, -1):
+        chi_left, d, chi_right = mps[i].shape
+        M = mps[i].reshape(chi_left, d * chi_right)
+
+        # Perform Singular Value Decomposition (SVD)
+        # M ≈ U @ diag(S) @ Vd, where:
+        # - U represents an orthogonal rotation in the space of chi_left
+        # - diag(S) scales the singular values (stretching or contracting)
+        # - Vd represents another rotation in the space of d * chi_right
+        U, S, Vd = qml.math.linalg.svd(M, full_matrices=False)
+
+        # Truncate SVD components if needed
+        chi_new = min(int(max_bond_dim), len(S))
+        U = U[:, :chi_new]
+        S = S[:chi_new]
+        Vd = Vd[:chi_new, :]
+
+        # Store Vd reshaped as an MPS tensor in the output MPS
+        output_mps[i] = Vd.reshape(chi_new, d, chi_right)
+
+        # Contract U with diag(S) to pass the information to the left
+        US = U @ qml.math.diag(S)
+
+        # Merge US with the preceding MPS tensor, preserving the canonical structure
+        mps[i - 1] = qml.math.tensordot(mps[i - 1], US, axes=([2], [0]))
+
+    output_mps[0] = mps[0]
+    return output_mps
+
+
 class MPSPrep(Operation):
     r"""Prepares an initial state from a matrix product state (MPS) representation.
 
     .. note::
 
-        Currently, this operator can only be used with ``qml.device(“lightning.tensor”)``.
+        This operator is natively supported on the ``lightning.tensor`` device, designed to run MPS structures
+        efficiently. For other devices, implementing this operation uses a gate-based decomposition which requires
+        auxiliary qubits (via ``work_wires``) to prepare the state vector represented by the MPS in a quantum circuit.
+
 
 
     Args:
-        mps (List[Array]):  list of arrays of rank-3 and rank-2 tensors representing an MPS state as a
-            product of site matrices. See the usage details section for more information.
+        mps (list[TensorLike]):  list of arrays of rank-3 and rank-2 tensors representing an MPS state
+            as a product of site matrices. See the usage details section for more information.
 
         wires (Sequence[int]): wires that the template acts on
+        work_wires (Sequence[int]): list of extra qubits needed in the decomposition. The maximum permissible bond
+            dimension of the provided MPS is defined as ``2^len(work_wires)``. Default is ``None``.
+        right_canonicalize (bool): Indicates whether a conversion to right-canonical form should be performed to the mps.
+            Default is ``False``.
+
+    The decomposition follows Eq. (23) in `[arXiv:2310.18410] <https://arxiv.org/pdf/2310.18410>`_.
+
+    A right canonicalization of the MPS is required for non-tensor devices.
```

Comment:
non-tensor devices means anything other than `lightning.tensor`? Any reason to not add this info to the note above?